api/coreservice.go

```diff
@@ -1227,10 +1231,18 @@ func (core *coreService) reverseActionsInBlock(blk *block.Block, reverseStart, c
 	blkHash := hex.EncodeToString(h[:])
 	blkHeight := blk.Height()
 
-	var res []*iotexapi.ActionInfo
-	for i := reverseStart; i < uint64(len(blk.Actions)) && i < reverseStart+count; i++ {
-		ri := uint64(len(blk.Actions)) - 1 - i
-		selp := blk.Actions[ri]
+	size := uint64(len(blk.Actions))
+	if reverseStart > size || count == 0 {
+		return nil
+	}
+	reserveEnd := reverseStart + count
+	if reserveEnd > size {
+		reserveEnd = size
+	}
+	res := make([]*iotexapi.ActionInfo, 0, reserveEnd-reverseStart)
+
+	for i := reserveEnd; i > reverseStart; i-- {
```

Comment:
the idx increases from `size-reserveEnd` to `size-reverseStart`?

Reply:
slice reverted in loop to reduce memory allocation.

Reply:
// start and end are calculated from `reverseStart` and `count`
for idx := start; idx < end; idx++ {
    selp := blk.Actions[idx];
    ...
}